FlowCrypt/Functionality/Mail Provider/Message Provider/MessageService.swift

```diff
@@ -278,35 +277,28 @@ final class MessageService {
 
 // MARK: - Message verification
 extension MessageService {
-    private func fetchVerificationPubKeys(for sender: String?) -> [String] {
-        if let sender = sender {
-            return contactsService.retrievePubKeys(for: sender)
-        } else {
-            return []
-        }
+    private func fetchVerificationPubKeys(for sender: String?) async throws -> [String] {
+        guard let sender = sender else { return [] }
+
+        let pubKeys = contactsService.retrievePubKeys(for: sender)
+        if pubKeys.isNotEmpty { return pubKeys }
+
+        guard let contact = try? await contactsService.searchContact(with: sender)
+        else { return [] }
+
+        return contact.activePubKeys.map(\.armored)
     }
```

Comment:
This really highlights the need to (later) be clearer in differentiating local and remote lookups. Because from the code, I can't tell. #830